### ISSUE TITLE
ApplyTypeAnnotationsVisitor does not consider function signatures

### ISSUE THREAD

**Issue by martindemello** [Instagram/LibCST#507]
merging the following files:
```
# py
def f(x, y): pass
```
and
```
# pyi
def f(y: int) -> int: ...
```
yields
```
def f(x, y: int) -> int: pass
```

It should have noted that the two functions differ in signature and not attempted to annotate anything.

Similarly, merging `def f(x): pass` with `def f(a: int) -> int: ...` should yield ether no change if we're matching names, or `def f(x: int) -> int` if we're matching posargs, but not `def f(x) -> int`.

**Comment by martindemello** [comment]
working on this

**Comment by martindemello** [comment]
fixed by #591
